### PR TITLE
Zstd compressed files

### LINKED ISSUE #4: Matcher: support compressed files
Support compressed files for the assertion matcher, which compares the expected and actual output.

In particular, the outputs of the API can be requested to be compressed, and also the expected results may be large, and should therefore be possible to be stored in compressed files

### PR DESCRIPTION
**Pull request by Taepper**
Resolves #4 

In order to keep the result file sizes tractable, we should not store them in a raw format in each of the cases. Instead, we add the option for result files to be decompressed before comparing the contents to the response.

### CODE REVIEW

**Review by corneliusroemer — APPROVED** [review]
I haven't tested but the overall direction looks right! Thanks!

**Comment by Taepper** [comment]
Yes, I think the compose file has a race-condition unfortunately :(

**Review by fengelniederhammer — CHANGES_REQUESTED** [review]
Let's separate this PR from the one that adapts the README?

File: test.ts [hunk 1]
```diff
@@ -20,6 +22,16 @@ function getTestsetDirectories() {
         .map((name) => path.join(testsetsPath, name));
 }
 
+async function getExpectedResponseString(file: string, compressed: boolean): Promise<string> {
```
[the remark this answers is not in the document]
Comment (fengelniederhammer) [reply test.ts:25]: I like to read files from top to bottom. When I start reading, I want to see either the starting point of execution or the high level stuff (preference on high level stuff). The further down it gets, the more details. This function here is some technical detail.

**Comment by corneliusroemer** [comment]
Right, your point is kind of this IIUC:
https://koroliov.com/2024/06/16/why-to-use-function-hoisting/

My question is just whether it matters at this point enough to require
changes, and whether this is the most important issue for review.

I could argue that if you hoist, then you encounter a function call but
it's a black box so anything after is undefined behavior. I'm not saying
that hoisting isn't worth it, just that this question is maybe not clearcut
enough to work on at this point.

**Comment by corneliusroemer** [comment]
You need to format at least :)

File: test.ts [hunk 2]
```diff
@@ -3,7 +3,9 @@ import { afterAll, beforeAll, describe, expect, it } from 'vitest';
 import * as path from 'path';
 import * as fs from 'fs';
 import { randomBytes } from 'node:crypto';
+import { decompress } from '@mongodb-js/zstd';
 
+// This number is used to start the distribution of ports to all the containers that are being run
```
Comment (fengelniederhammer) [inline test.ts:8]: IMO we don't need this comment

**Comment by corneliusroemer** [comment]
Woops, now CI fails as well. Locally never works for me btw. Would be good to get to the bottom of the flakiness with priority, it slows down everything else I think.

**Comment by fengelniederhammer** [comment]
I would also still prefer to seprate starting the containers and running the tests. Then this would also be quite a bit easier to debug.

File: test.ts (hunk 2 above)
Comment (Taepper) [reply test.ts:8]: I will move the definition closer to usage, which will make it clear enough I suppose

**Pull request merged**